CookingBook/src/networking/Server/Server.cpp

```diff
@@ -0,0 +1,219 @@
+#include "Server.h"
+
+CServer::CServer() :
+	ListeningSocket(INVALID_SOCKET),
+	sIpAddress(""),
+	iPort(0) ,
+	Time(std::time(0)),
+	CurrentTime(std::localtime(&Time)){
+	ZeroMemory(&ServerInfo, sizeof(ServerInfo));
+	iInfoSize = sizeof(ServerInfo);
+	ServerLogFile.open(g_sServerLogFileName,std::ios::binary|std::ios::app);
+}
+
+CServer::~CServer(){
+	WSACleanup();
+	closesocket(ListeningSocket);
+	if(ServerLogFile.is_open()){
+		ServerLogFile.close();
+	}
+}
+
+void CServer::LogInfo(const std::string& sInfo){
+	ServerLogFile<<"["<<CurrentTime->tm_year+1900<<":"<<
+	CurrentTime->tm_mon+1<<":"<<
+	CurrentTime->tm_mday<<":"<<
+	CurrentTime->tm_hour<<":"<<
+    CurrentTime->tm_min<<":"<<
+	CurrentTime->tm_sec<<"]"<<" "+sInfo<<std::endl;
+}
+
+void CServer::SetServer(const std::string& sIpAddress, const int iPort){
+	this->sIpAddress = sIpAddress;
+	this->iPort = iPort;
+	IPAddressFile.open(sIPAddressFileName,std::ios::binary|std::ios::out|std::ios::trunc);
+	IPAddressFile<<sIpAddress<<std::endl;
+	IPAddressFile<<iPort;
+	IPAddressFile.close();
+}
+
+bool CServer::CheckIP(){
+	IPAddressFile.open(sIPAddressFileName,std::ios::ate|std::ios::binary);
+	bool bResult;
+	if(IPAddressFile.is_open()){
+		if(IPAddressFile.tellg()==0){
```

Comment:
It returns the current position in the stream.
If either the stream buffer associated to the stream does not support the operation, or if it fails, the function returns -1.
streampos is an fpos type (it can be converted to/from integral types).